sunpy/map/maputils.py

```diff
@@ -188,10 +274,34 @@ def sample_at_coords(smap, coordinates):
     --------
     .. minigallery:: sunpy.map.sample_at_coords
     """
+
     if not all(contains_coordinate(smap, coordinates)):
-        raise ValueError('At least one coordinate is not within the bounds of the map.')
+        msg = "At least one coordinate is not within the bounds of the map."
+        raise ValueError(msg)
+
+    indices = np.array(smap.wcs.world_to_array_index(coordinates))
+
+    if method == "astropy":
+        return u.Quantity(smap.data[*indices], smap.unit)
+
+    extended_indices = indices.copy()
+    dimensions = [smap.dimensions.x.value, smap.dimensions.y.value]
+    for position in neighbor_positions:
+        extended_indices = np.append(
+            extended_indices,
+            get_neighboring_indices(indices, position, dimensions),
+            axis=1,
+        )
```

Comment:
You don't do anything to filter out duplicates...

Reply:
Good point. I wonder if it affects the higher-order methods. Will look further into this.

Reply:
The answer is no. But I'll filter duplicates since it will improve performance for `griddata`.